### PR TITLE
fix: use print() function in both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by cclauss**
#### Description of Change

notes: Fix issues that will halt the runtime on modern Python.

Legacy __print__ statements are ___syntax errors___ in Python 3 but __print()__ function works as expected in both Python 2 and Python 3. 

Old style exceptions are ___syntax errors___ in Python 3 but new style exceptions work as expected in both Python 2 and Python 3.

* __basestring__ was removed in Python 3 in favor of __str__.
* __file()__ was removed in Python 3 in favor of __open()__ which works in both Python 2 and Python 3.

225 days until Python 2 end of life.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes:

### CODE REVIEW

**Comment by nornagon** [comment]
I'm all for switching to Py3 but we very heavily depend on Py2 currently because Chromium's build scripts are all Py2. Are you aware of any efforts in Chromium to upgrade to 3?

File: script/pump.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
            | EMPTY
```
Comment (nornagon) [inline script/pump.py]: I think we don't actually use this script any more. cc @zcbenz who I think added it originally.

File: tools/linux/rewrite_dirs.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 # found in the LICENSE file.
```
Comment (nornagon) [inline tools/linux/rewrite_dirs.py]: I don't think we use this any more either.

**Comment by cclauss** [comment]
@nornagon None of these changes break Python 2 compatibility.  There has been a lot of movement towards Python 3 compatibility in the Chromium codebase in recent months.  Please check it if you get a chance.

**Comment by cclauss** [comment]
Am I correct in assuming that the __build-mac__ failure is unrelated?

**Comment by MarshallOfSound** [comment]
@cclauss Yup, that failure is due to the PR coming in from a fork.  That test always fails for forks at the moment --> #17668

File: script/pump.py (hunk 1 above)
Comment (codebytere) [reply script/pump.py]: we don't: @cclauss would you please remove it instead?

File: tools/linux/rewrite_dirs.py (hunk 2 above)
Comment (codebytere) [reply tools/linux/rewrite_dirs.py]: this gets used by `pkg-config-wrapper`, which i believe we still use to wrap around `pkg-config` to generate the correct include and  library paths when cross-compiling using a sysroot

Comment (nornagon) [reply tools/linux/rewrite_dirs.py]: I couldn't find any reference to this script, can you link me to where it's used?

Comment (codebytere) [reply tools/linux/rewrite_dirs.py]: cc @deepak1556 i think you may know more?

Comment (deepak1556) [reply tools/linux/rewrite_dirs.py]: This is a pre GN thing, not needed anymore.

**Comment by codebytere** [comment]
@cclauss can you resolve the merge conflicts? once that's done we can get this merged!

**Pull request merged**